src/v/kafka/server/group.cc

```diff
@@ -623,6 +623,9 @@ group::join_group_stages group::update_static_member_and_rebalance(
                         new_member_id,
                         std::move(md));
                       try_finish_joining_member(member, std::move(reply));
+                  })
+                  .then([group = shared_from_this()] {
```

Comment:
The common pattern is to use .finally()

Reply:
Oh yes, this and the next should be a finally though both work here I guess as there is no body, it's just a holder and it should stay around until the previous continuation is resoled regardless.